volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -351,17 +351,38 @@ def dereference(self) -> interfaces.objects.ObjectInterface:
 class DEVICE_OBJECT(objects.StructType, pool.ExecutiveObject):
     """A class for kernel device objects."""
 
-    def get_device_name(self) -> str:
-        header = self.get_object_header()
-        return header.NameInfo.Name.String  # type: ignore
-
+    def get_device_name(self) -> Union[str, interfaces.renderers.BaseAbsentValue]:
```

Comment:
I understand the reason behind this change, but we very explicitly allow exceptions to be raise up until the point of the plugin, so that the plugin can decide when/whether to catch the exception, rather than the framework.

Also, we only tend to use `BaseAbsentValues` within the plugin, because otherwise trying to use this in the core code means people will need to interpret results in some cases, None for other and now BaseAbsentValues in this case.  As it is, it's not a big deal because this should be throwing an exception, but for other situations returning a `BaseAbsentValue` should only really be done by plugin specific code (so not even shared `classmethod`s should be doing it).  It's strictly for rendering and results, not for internal use inside the code.  It will add more exception handling in other places, but at least then it all happens explicitly and everyone can see what gets caught, when and where...

Reply:
I understand what you are saying.
I felt that it was difficult to process errors even during actual work.
This method will return only pure values or objects, and errors that occur when the value is retrieved will be handled by the `_generator`. (There is a little bit more syntax for try and except. 😅)